lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -668,21 +671,7 @@ struct BubbleUpThroughFillOp : public OpRewritePattern<tensor::PackOp> {
       return failure();
 
     Value fillRes = fillOp.getResult(0);
-    auto users = fillRes.getUsers();
-
-    // Only two users: a pack and an unpack.
-    if (std::distance(users.begin(), users.end()) != 2)
-      return failure();
-
-    bool foundPack, foundUnPack = false;
-    for (Operation *op : users) {
-      if (isa_and_nonnull<tensor::UnPackOp>(op))
-        foundUnPack = true;
-      if (isa_and_nonnull<tensor::PackOp>(op))
-        foundPack = true;
-    }
-
-    if (!foundUnPack || !foundPack)
+    if (!fillRes.hasOneUse())
```

Comment:
I'm not sure that's equivalent to the check that was there before.

Sure, the fill has "more than one use" was encoded before, but also was that those need to be both pack and unpack (at least). Now, IIUC, any two users (even non-pack ones) will trigger this replacement?

Reply:
It is not equivalent. Now the fill result must have a single user (tensor.pack).

Reply:
No, I mean, before it was checking for a specific op (pack/unpack), now "any" op will do. What if `fill` has a user that is not `pack`?

Reply:
No, it can have only the pack. You get the fillOp by getting the pack operand, thus the result of the fillOp will be the pack.

Reply:
Of course! I need to click to expand the code more often, as Github hides too much of the context.